core/src/main/java/org/apache/druid/java/util/common/UnsafeUtils.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.java.util.common;
+
+import java.lang.reflect.Field;
+
+/**
+ * This utiliy class enables runtime access to sun.misc.Unsafe
+ *
+ * Reflection is required because this class is not available at compile time in Java 9 and above.
```

Comment:
This statement is not correct. `sun.misc.Unsafe` is still compile-time available in Java 9 and 10, but the access issues a warning in stderr. Since Java 11, it's still available, but only if `--add-modules jdk.unsupported` is added in runtime Java options.

Reply:
fine, I've made the comment less specific =